cranelift/assembler-x64/src/mem.rs

```diff
@@ -272,6 +273,28 @@ impl<R: AsReg, M: AsReg> GprMem<R, M> {
     }
 }
 
+/// An XMM register or memory operand.
+#[derive(Clone, Debug)]
+#[cfg_attr(any(test, feature = "fuzz"), derive(arbitrary::Arbitrary))]
+#[allow(
+    clippy::module_name_repetitions,
+    reason = "'XmmMem' indicates this has Xmm and memory variants"
+)]
+pub enum XmmMem<R: AsReg, M: AsReg> {
+    Xmm(R),
+    Mem(Amode<M>),
+}
+
+impl<R: AsReg, M: AsReg> XmmMem<R, M> {
+    /// Pretty-print the operand.
+    pub fn to_string(&self) -> String {
+        match self {
+            XmmMem::Xmm(xmm) => xmm::enc::to_string(xmm.enc()).to_owned(),
```

Comment:
```suggestion
            XmmMem::Xmm(xmm) => enc::to_string(xmm.enc()).to_owned(),
```

Reply:
I need `enc` from xmm module, similar to `let base = reg::enc::to_string(base.enc(), Size::Quadword);` needs enc from reg. I could just do `use crate::xmm::enc` but its easier to distinguish between `reg::enc` and `xmm::enc` this way I think.

Reply:
Oh, ok, I thought we were already in `xmm.rs` here. In that case, why not just use the existing `to_string` function?

```suggestion
            XmmMem::Xmm(xmm) => xmm.to_string(),
```

Reply:
Andrew and I had a chat, and we're going to leave this as is for now. There is some work to be done on `to_string` front in `api.rs` in `AsReg`, where `size` is an optional argument.